pkg/virt-handler/isolation/detector.go

```diff
@@ -143,20 +143,38 @@ func (s *socketBasedIsolationDetector) AdjustResources(vm *v1.VirtualMachineInst
 		if err != nil {
 			return err
 		}
-		rLimit := unix.Rlimit{
-			Max: uint64(memlockSize),
-			Cur: uint64(memlockSize),
-		}
-		err = prLimit(process.Pid(), unix.RLIMIT_MEMLOCK, &rLimit)
+
+		err = setProcessMemoryLockRLimit(process, memlockSize)
 		if err != nil {
-			return fmt.Errorf("failed to set rlimit for memory lock: %v", err)
+			return fmt.Errorf("failed to set process %d memlock rlimit to %d: %v", process.Pid(), memlockSize, err)
 		}
 		// we assume a single process should match
 		break
 	}
 	return nil
 }
 
+// setProcessMemoryLockRLimit Adjusts process MEMLOCK
+// soft-limit (current) and hard-limit (max) to the given size.
+func setProcessMemoryLockRLimit(process ps.Process, size int64) error {
```

Comment:
Please pass just the `pid` and not the whole `process`.

Reply:
Done